builtInFunctions/migrateCodeLeaf_test.go

```diff
@@ -0,0 +1,191 @@
+		t.Parallel()
+
+		rcl, _ := NewMigrateCodeLeafFunc(10, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{})
+		vmOutput, err := rcl.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), mock.NewUserAccount([]byte("dest")), nil)
+		assert.Nil(t, vmOutput)
+		assert.Equal(t, ErrNilVmInput, err)
+	})
+
+	t.Run("invalid num of args", func(t *testing.T) {
+		t.Parallel()
+
+		gasCost := uint64(10)
+
+		rcl, _ := NewMigrateCodeLeafFunc(gasCost, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{})
+
+		addr := []byte("addr")
+		key := []byte("codeHash")
+
+		vmInput := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallerAddr:  addr,
+				GasProvided: 50,
+				Arguments:   [][]byte{key},
+				CallValue:   big.NewInt(0),
+			},
+			RecipientAddr: addr,
+		}
+
+		vmOutput, err := rcl.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), mock.NewUserAccount([]byte("dest")), vmInput)
+		require.Nil(t, vmOutput)
+		require.True(t, errors.Is(err, ErrInvalidNumberOfArguments))
+	})
+
+	t.Run("should not call with value", func(t *testing.T) {
+		t.Parallel()
+
+		gasCost := uint64(10)
+		rcl, _ := NewMigrateCodeLeafFunc(gasCost, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{})
+
+		addr := []byte("addr")
+		key := []byte("codeHash")
+
+		vmInput := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallerAddr:  addr,
+				GasProvided: 50,
+				Arguments:   [][]byte{key},
+				CallValue:   big.NewInt(2),
+			},
+			RecipientAddr: addr,
+		}
+
+		vmOutput, err := rcl.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), mock.NewUserAccount([]byte("dest")), vmInput)
+		require.Nil(t, vmOutput)
+		require.True(t, errors.Is(err, ErrInvalidNumberOfArguments))
+	})
+
+	t.Run("should not call with value", func(t *testing.T) {
+		t.Parallel()
+
+		gasCost := uint64(10)
+		rcl, _ := NewMigrateCodeLeafFunc(gasCost, &mock.EnableEpochsHandlerStub{}, &mock.AccountsStub{})
+
+		addr := []byte("addr")
+
+		vmInput := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallerAddr:  addr,
+				GasProvided: 50,
+				CallValue:   big.NewInt(2),
+			},
+			RecipientAddr: addr,
+		}
+
+		vmOutput, err := rcl.ProcessBuiltinFunction(mock.NewUserAccount([]byte("sender")), mock.NewUserAccount([]byte("dest")), vmInput)
+		require.Nil(t, vmOutput)
+		require.Equal(t, ErrBuiltInFunctionCalledWithValue, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		wasCalled := false
+		accounts := &mock.AccountsStub{
+			GetCodeCalled: func(b []byte) []byte {
+				return []byte("key code")
+			},
+			MigrateCodeLeafCalled: func(account vmcommon.AccountHandler) error {
+				wasCalled = true
+				return nil
+			},
+		}
+
+		gasCost := uint64(10)
+
+		rcl, _ := NewMigrateCodeLeafFunc(gasCost, &mock.EnableEpochsHandlerStub{}, accounts)
+
+		addr := []byte("addr")
+
+		vmInput := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallerAddr:  addr,
+				GasProvided: 50,
```

Comment:
can you move this use a variable as well as for the expected remaining gas cost? so that it is more obvious

Reply:
:+1: